pytorch_transformers/modeling_utils.py

```diff
@@ -352,7 +357,28 @@ def prune_heads(self, heads_to_prune):
                 heads_to_prune: dict with keys being selected layer indices (`int`) and associated values being the list of heads to prune in said layer (list of `int`).
         """
         base_model = getattr(self, self.base_model_prefix, self)  # get the base model if needed
-        base_model._prune_heads(heads_to_prune)
+
+        to_be_pruned = {}
+
+        for layer, heads in heads_to_prune.items():
+            if int(layer) not in self.config.pruned_heads:
+                self.config.pruned_heads[int(layer)] = heads
```

Comment:
Here we have to be careful about one thing (that may be a questionable design choice): the configuration is stored in all the models that inherit from PreTrainedModel and thus a typical model with a base transformer and a head on top of it (like BertForXX) will have two configurations (one in the base model and one in the derived model).
I try to keep them in sync when they are updated after loading (which only happened when we resized the embeddings to add/remove tokens up to now).

Can you check with the `resize_embedding` method and do the same thing to keep the configurations in sync? Maybe we could also think about a better way to have a single configuration (but being able to always access `model.config` independant of the type of model is a nice feature).

Reply:
Sure I'll look into syncing the two configurations. I agree that having two configuration files is complicated, but as long as they're in sync it shouldn't be problematic.